integration/autoScalingGroup.go

```diff
@@ -0,0 +1,124 @@
+package integration
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/blang/semver"
+)
+
+// AutoScalingGroup represents an autoscaling group.
+type AutoScalingGroup struct {
+	Name      			string
+	Instances 			[]Instance
+	InstanceDetails InstanceDetails
+}
+
+func NewAutoScalingGroup(name string, instances []*autoscaling.Instance, instanceDetails InstanceDetails) AutoScalingGroup {
+	asg := AutoScalingGroup {
+		Name: name,
+		Instances: make([]Instance, len(instances)),
+		InstanceDetails: instanceDetails,
+	}
+
+	for i, awsInstance := range instances {
+		asg.Instances[i] = Instance {
+			ID:							aws.StringValue(awsInstance.InstanceId),
+			HealthStatus: 	aws.StringValue(awsInstance.HealthStatus),
+			LifecycleState: aws.StringValue(awsInstance.LifecycleState),
+		}
+	}
+
+	return asg
+}
+
+func (group AutoScalingGroup) GetTargetInstances(canonical semver.Version, minimumInstanceCount int) ([]string, error) {
+	start := time.Now()
+	healthy, unhealthy := categoriseInstances(group.Instances, minimumInstanceCount)
+
+  fmt.Printf("%s => %d healthy instances, %d unhealthy instances\n\thealthy: %+v\n\tunhealthy: %+v\n",
+    group.Name, len(healthy), len(unhealthy),
+    healthy,
+    unhealthy)
+
+  if len(healthy) <= minimumInstanceCount {
```

Comment:
The formatting is all wrong here. You should be using gofmt to format source code according to the language rules.